modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -471,6 +467,54 @@ private List<ObjectValidationRule> _getObjectValidationRules(
 		return objectValidationRules;
 	}
 
+	private List<ObjectValidationRuleSetting>
+			_overrideObjectValidationRuleSettings(
+				ObjectValidationRule objectValidationRule,
+				List<ObjectValidationRuleSetting> objectValidationRuleSettings)
+		throws PortalException {
+
+		Set<ObjectValidationRuleSetting> newObjectValidationRuleSettings =
+			SetUtil.fromCollection(objectValidationRuleSettings);
+
+		for (ObjectValidationRuleSetting existingObjectValidationRuleSetting :
+				_objectValidationRuleSettingPersistence.
+					findByObjectValidationRuleId(
+						objectValidationRule.getObjectValidationRuleId())) {
+
+			for (ObjectValidationRuleSetting objectValidationRuleSetting :
+					objectValidationRuleSettings) {
+
+				if (StringUtil.equals(
+						existingObjectValidationRuleSetting.getName(),
+						objectValidationRuleSetting.getName()) &&
+					StringUtil.equals(
+						existingObjectValidationRuleSetting.getValue(),
+						objectValidationRuleSetting.getValue())) {
+
+					newObjectValidationRuleSettings.remove(
+						objectValidationRuleSetting);
+				}
+				else {
+					_objectValidationRuleSettingPersistence.remove(
+						existingObjectValidationRuleSetting);
```

Comment:
@devpedrotavares you cannot remove inside this inner loop because you are only checking the first element of `objectValidationRuleSettings` and deleting the rest.

Imagine we have:
```
existingObjectValidationRuleSetting = old1
objectValidationRuleSettings = [new1, old1]
```

In this case `old1 != new1` and then we will remove the `old1` before checking if `old1 == old1`.

Reply:
My bad! I fixed it, could you check now?